### PR TITLE
Hello world

### PR DESCRIPTION
**Pull request by matt-paul**
The aim of this PR is to get a very simple app into our cluster.  This will allow us to move onto other tickets such as setting up our tracing solution.

- [x] Shamelessly lift app from [MHRA](https://github.com/MHRA/products).
- [x]  Create Github workflow to build and push image to Azure Container Registry.
- [x] Create temporay k8s deployment (deployments will live in separate repo at a later date but can live with the app for now.)

### CODE REVIEW

File: .github/workflows/hello-world.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        with:
+          path: platform
+
+      - name: "Login via Azure CLI"
+        uses: azure/login@v1
+        with:
+          creds: ${{ secrets.AZURE_CREDENTIALS }}
+      - name: "Login to docker"
+        uses: azure/docker-login@v1
+        with:
```
Comment (matt-paul) [inline .github/workflows/hello-world.yml:31]: Standing on the shoulders of giants here, thanks Tracy!

File: .github/workflows/hello-world.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+on:
+  push:
+    branches:
+      - "**"
+    paths:
+      - hello-world/**
+      - .github/workflows/hello-world.yml
+
+env:
+  IMAGE: citracytest.azurecr.io/platform/hello-world
```
Comment (matt-paul) [inline .github/workflows/hello-world.yml:12]: As mentioned, we can either merge this in and change the container registry later once set up, or hold off merging this PR

File: hello-world/Makefile [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+.PHONY: docker-build
+docker-build: ## Build and tag Docker image
+	DOCKER_BUILDKIT=1 docker build \
+		--progress plain \
+		--tag $(app) \
+		--tag $(image) \
+		--tag $(image):$(tag) \
+		.
+.PHONY: docker-run
```
Comment (frigus02) [inline hello-world/Makefile:21]: There should probably be a empty line before this

File: hello-world/Cargo.toml [hunk 4]
```diff
@@ -0,0 +1,15 @@
+[package]
+name = "hello-world"
+version = "0.1.0"
+authors = ["stuart.harris@red-badger.com"]
```
Comment (alsuren) [inline hello-world/Cargo.toml:4]: This entire folder smells like copy-pasta. Can you add a README explaining where you got it from, and/or attribution comments at the top of each file? You may also need to copy a LICENSE file into this directory.

File: .gitignore [hunk 5]
```diff
@@ -1,3 +1,5 @@
 .env
 .terraform/
 terraform.tfvars
+
+target
```
Comment (frigus02) [inline .gitignore:5]: That's a folder, isn't it? Should we add a slash to make this easier to understand? `target/`

File: hello-world/Cargo.toml (hunk 4 above)
Comment (matt-paul) [reply hello-world/Cargo.toml:4]: Hmm - i deliberately left Stu here because of that.  That's not enough attribution? Good shout about the README though

Comment (matt-paul) [reply hello-world/Cargo.toml:4]: ah, see it's 'and / or' 👍

File: hello-world/Makefile [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+	docker run -it -p 3030:3030 $(image)
+
+.PHONY: docker-push
+docker-push: ## Push Docker image
+	docker push $(image)
+	docker push $(image):$(tag)
+
+.PHONY: help
+help: ## Display this help screen
+	@grep -E '^[a-zA-Z_-]+:.*?## .*$$' $(MAKEFILE_LIST) | sort | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m%-30s\033[0m %s\n", $$1, $$2}'
```
Comment (alsuren) [inline hello-world/Makefile:32]: maybe chop this line up with \ like how line 15 is done? (also add attribution? Feels quite similar to https://gist.github.com/prwhite/8168133#gistcomment-3291344 )

File: hello-world/Cargo.toml (hunk 4 above)
Comment (matt-paul) [reply hello-world/Cargo.toml:4]: As discussed, will add the MIT License from MHRA into this directory and add a note in a README.

**Pull request merged**